src/signature.c

```diff
@@ -260,12 +283,14 @@ static bool get_pubkey(void)
 	cert = PEM_read_X509(fp_pubkey, NULL, NULL, NULL);
 	if (!cert) {
 		fprintf(stderr, "Failed PEM_read_X509() for %s\n", CERTNAME);
+		fp_pubkey = NULL;
```

Comment:
Don't we hit a file descriptor leak by setting these to NULL? `terminate_signature` will only call `fclose` if `fp_pubkey` is not NULL.